### PR TITLE
Fix portal deletion

### PR DESCRIPTION
**Pull request by sunny-viktoryia**
fixes https://github.com/sparkletown/internal-sparkle-issues/issues/1803

### CODE REVIEW

File: src/components/admin/PortalAddEditForm.tsx [hunk 1]
```diff
@@ -170,13 +170,13 @@ export const PortalAddEditForm: React.FC<PortalAddEditFormProps> = ({
...
-    deletePortal,
+    removePortal,
   ] = useAsyncFn(async () => {
     if (!currentSpaceId || !portal) return;
 
-    await deleteRoom(currentSpaceId, portal);
+    await deletePortal(currentSpaceId, portal);
     await onDone();
-  });
+  }, [currentSpaceId, onDone, portal]);
```
Comment (sunny-viktoryia) [inline src/components/admin/PortalAddEditForm.tsx:179]: the fix is here

File: functions/src/venue.ts [hunk 2]
```diff
@@ -598,36 +598,36 @@ export const upsertRoom = functions.https.onCall(async (data, context) => {
...
-    spaceId: venueId,
+    spaceId,
     worldId: space.worldId,
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
     userId: context.auth.token.user_id,
   });
 
-  const doc = await admin.firestore().collection("venues").doc(venueId).get();
+  const doc = await admin.firestore().collection("venues").doc(spaceId).get();
```
Comment (denisdimitrov) [inline functions/src/venue.ts:612]: The space is already fetched at line 606, this line can be removed and make the request complete faster.

File: functions/src/venue.ts [hunk 3]
```diff
@@ -598,36 +598,36 @@ export const upsertRoom = functions.https.onCall(async (data, context) => {
...
-  const space = await getSpaceById(venueId);
+  const space = await getSpaceById(spaceId);
 
   await throwErrorIfNeitherWorldNorSpaceOwner({
-    spaceId: venueId,
+    spaceId,
     worldId: space.worldId,
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
     userId: context.auth.token.user_id,
```
Comment (denisdimitrov) [inline functions/src/venue.ts]: [OOS] 
```suggestion
    userId: context.auth?.token.user_id,
```

**Review by alexxpetrov — APPROVED** [review]
LGTM

File: src/components/admin/PortalAddEditForm.tsx (hunk 1 above)
Comment (denisdimitrov) [reply src/components/admin/PortalAddEditForm.tsx:179]: @sunny-viktoryia The same code exists in `ScreeningRoomVideoAddEditForm.tsx` 

[OOS] Not sure if that code is still relevant but we might have to fix that one as well.

Comment (sunny-viktoryia) [reply src/components/admin/PortalAddEditForm.tsx:179]: great catch! I'll check.

Comment (sunny-viktoryia) [reply src/components/admin/PortalAddEditForm.tsx:179]: done: https://github.com/sparkletown/sparkle/pull/3000/commits/5ebf45d88e857b548d00b79459c89244c3d6e92e

File: functions/src/venue.ts (hunk 2 above)
Comment (sunny-viktoryia) [reply functions/src/venue.ts:612]: done: https://github.com/sparkletown/sparkle/pull/3000/commits/5ebf45d88e857b548d00b79459c89244c3d6e92e

File: functions/src/venue.ts (hunk 3 above)
Comment (sunny-viktoryia) [reply functions/src/venue.ts]: done: https://github.com/sparkletown/sparkle/pull/3000/commits/5ebf45d88e857b548d00b79459c89244c3d6e92e

File: functions/src/venue.ts [hunk 4]
```diff
@@ -598,36 +596,34 @@ export const upsertRoom = functions.https.onCall(async (data, context) => {
...
-  const doc = await admin.firestore().collection("venues").doc(venueId).get();
+  const doc = await admin.firestore().collection("venues").doc(spaceId).get();
 
   if (!doc || !doc.exists) {
-    throw new HttpsError("not-found", `Venue ${venueId} not found`);
+    throw new HttpsError("not-found", `Venue ${spaceId} not found`);
   }
   const docData = doc.data();
   if (!docData) {
     throw new HttpsError("internal", `Data not found`);
```
Comment (goran-peoski-work) [inline functions/src/venue.ts]: [NOTE]
Interesting, the above has `not-found` code, this one just `internal`. Might be an older line or copy-paste thing that makes them different. Might as well make them all `not-found`

**Review by goran-peoski-work — APPROVED** [review]
LGTH

File: functions/src/venue.ts (hunk 4 above)
Comment (sunny-viktoryia) [reply functions/src/venue.ts]: changed to "not-found": https://github.com/sparkletown/sparkle/pull/3000/commits/7051adca367c073619ecacf694eaa4bc4656cd4c

**Pull request merged**